src/browser/transports/unix_domain_socket.ts

```diff
@@ -0,0 +1,84 @@
+import { app } from 'electron';
+import { unlink } from 'fs';
+import { exec } from 'child_process';
+
+// we need to use require here because unix-dgram is an optional dependency
+// and is only available on unix systems. import does a static check and
+// will not compile if the module isn't present at time of compilation
+const unixDgram = process.platform === 'win32' ? {} : require('unix-dgram');
+
+import BaseTransport from './base';
+
+type FileDescriptor = string;
+
+interface Socket {
+    close: () => void;
+    bind: (name: string) => void;
+    on: (event: string, callback?: (...args: any[]) => void) => void;
+    send: (buffer: Buffer, offset: number, length: number, path: FileDescriptor) => void;
+}
+
+class UnixDomainSocket extends BaseTransport {
+    private fileDescriptorPrefix: string;
+    private serverName: FileDescriptor;
+    private server: Socket;
+
+    constructor(fileDescriptorPrefix: string) {
+        super();
+
+        this.fileDescriptorPrefix = fileDescriptorPrefix;
+        this.serverName = fileDescriptorPrefix + Date.now();
+        this.server = unixDgram.createSocket('unix_dgram', (buffer: Buffer) => {
+            this.eventEmitter.emit('message', null, buffer);
+        });
+        this.server.bind(this.serverName);
+
+        app.on('will-quit', () => {
+            this.cleanUpServer();
+        });
+
+        process.on('SIGINT', () => {
+            this.cleanUpServer();
+        });
+    }
+
+    public publish(data: any): boolean {
+        const message = new Buffer(JSON.stringify(data));
+        this.getFileDescriptors().then((fds: FileDescriptor[]) => {
```

Comment:
On `publish()`, we'll look for all open file descriptors that contain `this.fileDescriptorPrefix` and create a client on the fly to send over the data. Once sent, the client will close.